test/torch/tensors/test_promise.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+import torch
+import syft
+
+
+def test__str__():
+    a = syft.Promises.FloatTensor(shape=torch.Size((3, 3)))
+    assert isinstance(a.__str__(), str)
+
+
+@pytest.mark.parametrize("cmd", ["__add__", "sub", "__mul__"])
+def test_operations_between_promises(hook, cmd):
+    hook.local_worker.is_client_worker = False
+
+    a = syft.Promises.FloatTensor(shape=torch.Size((2, 2)))
+    b = syft.Promises.FloatTensor(shape=torch.Size((2, 2)))
+
+    actual = getattr(a, cmd)(b)
+
+    ta = torch.tensor([[1.0, 2], [3, 4]])
+    tb = torch.tensor([[-8.0, -7], [6, 5]])
+    a.keep(ta)
+    b.keep(tb)
+
+    expected = getattr(ta, cmd)(tb)
+
+    assert (actual.value() == expected).all()
+
+    hook.local_worker.is_client_worker = True
+
+
+@pytest.mark.parametrize("cmd", ["__add__", "sub", "__mul__"])
+def test_operations_with_concrete(hook, cmd):
+    hook.local_worker.is_client_worker = False
+
+    a = syft.Promises.FloatTensor(shape=torch.Size((2, 2)))
+    b = torch.tensor([[-8.0, -7], [6, 5]]).wrap()  # TODO fix need to wrap
+
+    actual = getattr(a, cmd)(b)
+
+    ta = torch.tensor([[1.0, 2], [3, 4]]).wrap()
```

Comment:
It is unfortunate that you have to wrap this. We previously haven't had good support when two tensors are interacting with each other where one has a different class hierarchy than the other. @LaRiffle any ideas?

Reply:
So, we have done specific stuff to allow AST to operate with FPT in specific cases, but nothing general
I don't really get why calling .wrap() here make the thing work :/

Reply:
I implemented a basic version of getting things to interact which dont' have the exact same class path. Not sure if it would work in this case but just FYI

Reply:
I implemented a basic version of getting things to interact which dont' have the exact same class path. Not sure if it would work in this case but just FYI